src/main/java/org/jfrog/bamboo/config/JfrogServerConfigAction.java

```diff
@@ -0,0 +1,226 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jfrog.bamboo.config;
+
+import com.atlassian.bamboo.ww2.BambooActionSupport;
+import com.atlassian.bamboo.ww2.aware.permissions.GlobalAdminSecurityAware;
+import org.apache.commons.lang.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jetbrains.annotations.NotNull;
+import org.jfrog.bamboo.utils.BuildLog;
+import org.jfrog.build.client.ArtifactoryVersion;
+import org.jfrog.build.extractor.clientConfiguration.client.artifactory.ArtifactoryManager;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+public class JfrogServerConfigAction extends BambooActionSupport implements GlobalAdminSecurityAware {
+
+    private final transient Logger log = LogManager.getLogger(JfrogServerConfigAction.class);
+
+    private String mode;
+    private String serverId;
+    private String url;
+    private String username;
+    private String password;
+    private String accessToken;
+    private String isSendTest;
+
+    private final transient ServerConfigManager serverConfigManager;
+
+    public JfrogServerConfigAction(ServerConfigManager serverConfigManager) {
+        this.serverConfigManager = serverConfigManager;
+        mode = "add";
+    }
+
+    @Override
+    public void validate() {
+        clearErrorsAndMessages();
+        if (StringUtils.isBlank(serverId)) {
+            addFieldError("serverId", "Please specify a Server ID identifier.");
+        } else if (mode.equals("add") && serverConfigManager.getServerConfigById(serverId) != null) {
+            addFieldError("serverId", "Server ID already exists.");
+        }
+
+        if (StringUtils.isBlank(url)) {
+            addFieldError("url", "Please specify a URL of a JFrog Platform.");
+        } else {
+            try {
+                new URL(url);
+            } catch (MalformedURLException mue) {
+                addFieldError("url", "Please specify a valid URL of a JFrog Platform.");
+            }
+        }
+
+        if (StringUtils.isNotBlank(username) && StringUtils.isBlank(password)){
+            addFieldError("password", "Please specify the password of your JFrog Platform.");
+        } else if (StringUtils.isBlank(username) && StringUtils.isNotBlank(password)){
+            addFieldError("username", "Please specify the username of your JFrog Platform.");
+        }
+    }
+
+    public String doAdd() throws Exception {
```

Comment:
Is this method in use? If so, please suppress the unused warning
```suggestion
    public String doAdd() {
```
**Same comment for all methods below in this file**

Reply:
yes